sql/driver/wire.go

```diff
@@ -80,14 +100,17 @@ func (d Datum) Value() (driver.Value, error) {
 		val = t.BytesVal
 	case *Datum_StringVal:
 		val = t.StringVal
+	case *Datum_DateVal:
+		val = Date{t.DateVal.GoTime()}
```

Comment:
The truncation in `Time.Truncate` is zone independent (i.e. always UTC), but the "truncation" implied by d.Format("2006-01-02") is not. http://play.golang.org/p/bzvgCPfD8a